code/game/gamemodes/umbrage/umbrage.dm

```diff
@@ -0,0 +1,184 @@
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain")
+	announce_span = "velvet"
+	announce_text = "Eldritch aberrations are trying to enslave the station!\n\
+	<span class='velvet'>Umbrages</span>: Dominate the will of the crew and ascend into a progenitor.\n\
+	<span class='notice'>Crew</span>: Protect your minds and nullify the umbrages before they take over."
+	var/list/initial_umbrages = list()
+
+/datum/game_mode/umbrage/pre_setup()
+	if(config.protect_roles_from_antagonist)
+		restricted_jobs += protected_jobs
+	if(config.protect_assistant_from_antagonist)
+		restricted_jobs += "Assistant"
+	var/starting_umbrages = Clamp(round(num_players() / 10), 1, 3) //At least 1 umbrage, but no more than 3
+	for(var/i = starting_umbrages, i > 0, i--)
+		var/datum/mind/new_umbrage = pick(antag_candidates)
+		initial_umbrages += new_umbrage
+		antag_candidates -= new_umbrage
+		modePlayer += new_umbrage
+		new_umbrage.special_role = "Umbrage"
+		new_umbrage.restricted_roles = restricted_jobs
+		log_game("[new_umbrage.key] (ckey) has been selected as an umbrage.")
+	return 1
+
+/datum/game_mode/umbrage/post_setup()
+	for(var/U in initial_umbrages)
+		var/datum/mind/umbrage_mind = U
+		var/mob/living/H = umbrage_mind.current
+		antag_umbrage(H, 1)
+		greet_umbrage(H)
+		equip_umbrage(H)
+	..()
+	return 1
+
+/datum/game_mode/proc/greet_umbrage(mob/living/U) //Announcement and direction to the tutorial
+	if(!U)
+		return
+	U << "<span class='velvet big'><b>You are an umbrage!</b></span>"
+	U << "<i>Use <b>.a</b> before your messages to speak over the Mindlink. This only works across your current z-level.</i>"
+	U << "<i>Look for the info button in the top left of your screen if you need help.</i>"
+	return 1
+
+/datum/game_mode/proc/greet_veil(mob/living/V)
+	if(!V)
+		return
+	V << "<span class='velvet big'><b>ukq wna ieja jks</b></span>" //"you are mine now"
+	V << "<b>Your mind goes numb. Your thoughts go blank. You feel utterly empty. \n\
+	A mind brushes against your own. You dream.\n\
+	Of a vast, empty Void in the deep of space.\n\
+	Something lies in the Void. Ancient. Unknowable. It watches you with hungry eyes. \n\
+	Eyes filled with stars.\n\
+	You feel a vast consciousness slowly consume your own and the veil falls away.\n\
+	Serve the umbrages above all else. Your former allegiances are now forfeit. Their goal is yours, and yours is theirs.</b>"
+	V << "<i>Use <b>.a</b> before your messages to speak over the Mindlink. This only works across your current z-level.</i>"
+	V << "<i>Ask for help from your masters or fellows if you're new to this role.</i>"
+	V << sound('sound/magic/become_veil.ogg', volume = 50)
+	flash_color(V, flash_color = "#21007F", flash_time = 100)
+
+/datum/game_mode/proc/equip_umbrage(mob/living/U)
+	var/datum/umbrage/S = new
+	S.linked_mind = U.mind
+	U.mind.umbrage_psionics = S
+	S.link_to_new_body(U)
+	S.give_ability("howto", 1)
+	S.give_ability("divulge", 1)
+	return
```

Comment:
Trailing returns are the ones that need a `return ..()` at the end, right? Or am I just a dummy?